### ISSUE TITLE
Smarter Client side decoration detection

### ISSUE THREAD

**Issue by fenryxo** [tiliado/diorite#2]
It is necessary to improve detection of support for client side decorations. E.g. get window manager name:

* [`wmctrl -m`](http://askubuntu.com/a/92948/51082)
* [`xprop -id $(xprop -root _NET_SUPPORTING_WM_CHECK | cut -d\# -f2)`](http://unix.stackexchange.com/a/20372/21415)

From https://groups.google.com/d/msg/nuvola-player-devel/5FrIQcrOOQA/gEvZok8TuloJ

>> On Wednesday, June 24, 2015 at 6:19:00 PM UTC+2, Jonas Frei wrote:
>> From what I understand, nuvolaplayer should detect automatically whether client side decorations are supported and should enable it in that case.

> That is not accurate. Nuvola Player (libdiorite) doesn't check whether client side decorations are supported, but it tries to guess whether it is running under GNOME Shell and then uses client side decorations.
> https://github.com/tiliado/diorite/blob/master/src/gtk/ApplicationWindow.vala#L72 

How to help
---------------

Run command `xprop -id $(xprop -root _NET_SUPPORTING_WM_CHECK | cut -d\# -f2)` in terminal and post here the output and your distribution & desktop environment.

### Debian Jessie - GNOME Shell 3.14.4

```
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x1600003
_MUTTER_VERSION(UTF8_STRING) = "3.14.4"
_GNOME_WM_KEYBINDINGS(UTF8_STRING) = "Mutter,GNOME Shell"
_NET_WM_NAME(UTF8_STRING) = "GNOME Shell"
```

**Comment by freijon** [comment]
### Arch Linux - GNOME Shell 3.16.3

```
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0xe00004
_MUTTER_VERSION(UTF8_STRING) = "3.16.3"
_GNOME_WM_KEYBINDINGS(UTF8_STRING) = "Mutter,GNOME Shell"
_NET_WM_NAME(UTF8_STRING) = "GNOME Shell"
```

**Comment by fenryxo** [comment]
Fixed in git master & 0.1.x branches.

**Comment by theblacklion** [comment]
elementary OS Freya - pantheon-shell 1.377+424~daily~ubuntu0.3.1 

```
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x120000b
_MUTTER_VERSION(UTF8_STRING) = "3.12.2"
_GNOME_WM_KEYBINDINGS(UTF8_STRING) = "Mutter"
_NET_WM_NAME(UTF8_STRING) = "Mutter(Gala)"
```
